web_src/js/features/notification.js

```diff
@@ -19,21 +19,53 @@ export function initNotificationsTable() {
 }
 
 export function initNotificationCount() {
-  if (NotificationSettings.MinTimeout <= 0) {
+  const notificationCount = $('.notification_count');
+
+  if (notificationCount.length <= 0) {
     return;
   }
 
-  const notificationCount = $('.notification_count');
+  if (NotificationSettings.EventSourceUpdateTime > 0 && !!window.EventSource) {
+    // Try to connect to the event source first
+    const source = new EventSource(`${AppSubUrl}/user/events`);
+    source.addEventListener('notification-count', async (e) => {
+      if (e.origin !== window.location.origin) {
```

Comment:
I don't know a thing about event sources but... can this really happen?

Reply:
yes

Reply:
I think this check should be removed. EventSource should already be protected by the same-origin policy and by removing it, we allow users to configure CORS to make this work cross-origin.

Reply:
https://html.spec.whatwg.org/multipage/web-messaging.html#authors

Reply:
OK on further investigation people disagree with that assertion as it only comes from https://www.html5rocks.com/en/tutorials/eventsource/basics/ so I've removed the checks.

Reply:
From what I know from websockets, it is crucial to perform the origin check on the server-side to prevent third-party origins from connecting to the socket. I think this is what your link refers to.

Reply:
Additionally, it may also be good to send the CSRF token along in the payload.

Reply:
So there's no actual payload for this - it's essentially a long-running GET.